rust/lancedb/src/remote/table.rs

```diff
@@ -115,39 +115,14 @@ impl<S: HttpSend> RemoteTable<S> {
     async fn read_arrow_stream(
         &self,
         request_id: &str,
-        body: reqwest::Response,
+        response: reqwest::Response,
     ) -> Result<SendableRecordBatchStream> {
-        // Assert that the content type is correct
-        let content_type = body
-            .headers()
-            .get(CONTENT_TYPE)
-            .ok_or_else(|| Error::Http {
-                source: "Missing content type".into(),
-                request_id: request_id.to_string(),
-                status_code: None,
-            })?
-            .to_str()
-            .map_err(|e| Error::Http {
-                source: format!("Failed to parse content type: {}", e).into(),
-                request_id: request_id.to_string(),
-                status_code: None,
-            })?;
-        if content_type != ARROW_STREAM_CONTENT_TYPE {
-            return Err(Error::Http {
-                source: format!(
-                    "Expected content type {}, got {}",
-                    ARROW_STREAM_CONTENT_TYPE, content_type
-                )
-                .into(),
-                request_id: request_id.to_string(),
-                status_code: None,
-            });
-        }
```

Comment:
Do we not verify the content-type any longer?

Reply:
See this issue: https://github.com/lancedb/sophon/issues/2742

The server returned the incorrect content type. None of the existing clients check this right now, so I think it's best for now we remove the check.